### DISCUSSION TITLE
Add column to all database tables as predefined and append it as index to all indexes to be defined?

### DISCUSSION THREAD

**Discussion by IbrahimFathy19** [prisma/prisma#15597] [Q&A] [answered]
Currently I'm implementing a multi-tenant solution using single shared database by defining A tenant column in each database table.
I'm thinking of automating this process by adding a tenant column by default to each and every table on creation, and the tenant as index compounded with any defined index in the table.

I tried to create a custom generator and adjust the `options.dmmf.datamodel` models, and I was thinking to overwrite the current schema with the new one. But somehow I can't generate a schema from `options.dmmf.datamodel`.

So, is there a function to convert `options.dmmf.datamodel` to string?
Is this even the right way to do so?

Another Solution would be create an event trigger ON CREATE TABLE and create that column, but this will make prisma migration out of sync with database!

**Comment by nurul3101** [comment, marked as answer]
Hey @IbrahimFathy19 👋 
What you are looking for is similar to [Inheritance](https://github.com/prisma/prisma/issues/2506) where you want to inherit the tenant column in every table. If you could leave a 👍 to the feature request then it would help us in prioritising it.
For now, you would need to manually add the tenant column in new tables.

Please note that as DMMF is an internal API it might contain breaking changes without prior updates.

**Reply by vladi-stevanovic** [reply]
👋 Hello @IbrahimFathy19 - did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions, or feel free to "Mark as answer" if this discussion is now solved!

**Reply by IbrahimFathy19** [reply]
@nurul3101 @vladi-stevanovic yes thank you. but actually I did a workaround and created a custom generator that updates the schema file based on my logic, and this [Package](https://github.com/prisma/upgrade) was really helpful.

Thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
